Failure to startup on GKE 1.29 nodes, worked on GKE 1.27 nodes

## monero container logs

```
kubectl logs -n support support-cryptnono-g7xcd -c monero

definitions.h:2:10: fatal error: 'asm/fpu/internal.h' file not found
```

## Other details

<details><summary>fetch-kernel-headers init-container logs (GKE 1.27)</summary>

```
+ LSB_FILE=/etc/lsb-release.host
+ OS_RELEASE_FILE=/etc/os-release.host
+ TARGET_DIR=/usr/src
+ HOST_MODULES_DIR=/lib/modules.host
++ uname -r
+ KERNEL_VERSION=5.15.120+
+ [[ ! -e /lib/modules/.installed ]]
Headers already installed
+ echo 'Headers already installed'
+ exit 0
```

</details>

<details><summary>fetch-kernel-headers init-container logs (GKE 1.29)</summary>

```
kubectl logs -n support support-cryptnono-g7xcd -c fetch-kernel-headers

+ LSB_FILE=/etc/lsb-release.host
+ OS_RELEASE_FILE=/etc/os-release.host
+ TARGET_DIR=/usr/src
+ HOST_MODULES_DIR=/lib/modules.host
++ uname -r
+ KERNEL_VERSION=6.1.58+
+ [[ ! -e /lib/modules/.installed ]]
+ check_headers /lib/modules.host
+ modules_path=/lib/modules.host
++ uname -m
+ arch=x86_64
+ kdir=/lib/modules.host/6.1.58+
+ [[ x86_64 == \x\8\6\_\6\4 ]]
+ arch=x86
+ [[ ! -e /lib/modules.host/6.1.58+ ]]
+ return 1
+ install_headers
++ awk '/^NAME *= */ { gsub(/^NAME *= */, ""); print }' /etc/os-release.host
+ distro='"Container-Optimized OS"'
+ case $distro in
+ install_cos_linux_headers
+ grep -q CHROMEOS_RELEASE_VERSION /etc/lsb-release.host
++ awk '/CHROMEOS_RELEASE_VERSION *= */ { gsub(/^CHROMEOS_RELEASE_VERSION *= */, ""); print }' /etc/lsb-release.host
+ BUILD_ID=17800.66.78
+ BUILD_DIR=/linux-lakitu-17800.66.78
+ SOURCES_DIR=/usr/src/linux-lakitu-17800.66.78
+ [[ ! -e /usr/src/linux-lakitu-17800.66.78/.installed ]]
+ mkdir -p /usr/src/linux-lakitu-17800.66.78
+ echo 'Installing kernel headers for COS build 17800.66.78'
+ fetch_cos_linux_sources
+ echo 'Fetching upstream kernel sources.'
+ mkdir -p /linux-lakitu-17800.66.78
Installing kernel headers for COS build 17800.66.78
Fetching upstream kernel sources.
+ curl -s https://storage.googleapis.com/cos-tools/17800.66.78/kernel-src.tar.gz
+ tar -xzf - -C /linux-lakitu-17800.66.78

real	0m49.415s
user	0m14.632s
sys	0m17.111s
+ generate_headers
+ echo 'Generating kernel headers'
+ cd /linux-lakitu-17800.66.78
+ '[' -e /proc/config.gz ']'
+ zcat /proc/config.gz
+ make ARCH=x86 oldconfig
Generating kernel headers

Error in reading or end of file.

Error in reading or end of file.

Error in reading or end of file.

Error in reading or end of file.

Error in reading or end of file.

Error in reading or end of file.

Error in reading or end of file.

Error in reading or end of file.

Error in reading or end of file.
+ make ARCH=x86 prepare
+ find /linux-lakitu-17800.66.78 -regex '.*\.c\|.*\.txt\|.*Makefile\|.*Build\|.*Kconfig' -type f -delete

real	1m10.120s
user	0m39.503s
sys	0m11.217s
+ rm -rf /usr/src/linux-lakitu-17800.66.78

real	0m0.001s
user	0m0.000s
sys	0m0.001s
+ mv /linux-lakitu-17800.66.78 /usr/src

real	0m5.460s
user	0m0.861s
sys	0m3.311s
+ touch /usr/src/linux-lakitu-17800.66.78/.installed
+ HEADERS_TARGET=/usr/src/linux-lakitu-17800.66.78
+ mkdir -p /lib/modules/6.1.58+
+ ln -sf /usr/src/linux-lakitu-17800.66.78 /lib/modules/6.1.58+/source
+ ln -sf /usr/src/linux-lakitu-17800.66.78 /lib/modules/6.1.58+/build
+ touch /lib/modules/.installed
+ exit 0
```

</details>

<details><summary>kubectl describe node (node where failure was seen)</summary>

```
kubectl describe node gke-qcl-cluster-core-pool-f96ea250-oiq6
Name:               gke-qcl-cluster-core-pool-f96ea250-oiq6
Roles:              <none>
Labels:             beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/instance-type=n2-highmem-2
                    beta.kubernetes.io/os=linux
                    cloud.google.com/gke-boot-disk=pd-balanced
                    cloud.google.com/gke-container-runtime=containerd
                    cloud.google.com/gke-cpu-scaling-level=2
                    cloud.google.com/gke-logging-variant=DEFAULT
                    cloud.google.com/gke-max-pods-per-node=110
                    cloud.google.com/gke-netd-ready=true
                    cloud.google.com/gke-nodepool=core-pool
                    cloud.google.com/gke-os-distribution=cos
                    cloud.google.com/gke-provisioning=standard
                    cloud.google.com/gke-stack-type=IPV4
                    cloud.google.com/machine-family=n2
                    cloud.google.com/private-node=false
                    failure-domain.beta.kubernetes.io/region=europe-west1
                    failure-domain.beta.kubernetes.io/zone=europe-west1-d
                    hub.jupyter.org/node-purpose=core
                    iam.gke.io/gke-metadata-server-enabled=true
                    k8s.dask.org/node-purpose=core
                    kubernetes.io/arch=amd64
                    kubernetes.io/hostname=gke-qcl-cluster-core-pool-f96ea250-oiq6
                    kubernetes.io/os=linux
                    node.kubernetes.io/instance-type=n2-highmem-2
                    node.kubernetes.io/masq-agent-ds-ready=true
                    projectcalico.org/ds-ready=true
                    topology.gke.io/zone=europe-west1-d
                    topology.kubernetes.io/region=europe-west1
                    topology.kubernetes.io/zone=europe-west1-d
Annotations:        container.googleapis.com/instance_id: 2156677655658839173
                    csi.volume.kubernetes.io/nodeid:
                      {"pd.csi.storage.gke.io":"projects/qcl-hub/zones/europe-west1-d/instances/gke-qcl-cluster-core-pool-f96ea250-oiq6"}
                    node.alpha.kubernetes.io/ttl: 0
                    node.gke.io/last-applied-node-labels:
                      cloud.google.com/gke-boot-disk=pd-balanced,cloud.google.com/gke-container-runtime=containerd,cloud.google.com/gke-cpu-scaling-level=2,clou...
                    node.gke.io/last-applied-node-taints: 
                    volumes.kubernetes.io/controller-managed-attach-detach: true
CreationTimestamp:  Thu, 09 May 2024 10:00:08 +0200
Taints:             <none>
Unschedulable:      false
Lease:
  HolderIdentity:  gke-qcl-cluster-core-pool-f96ea250-oiq6
  AcquireTime:     <unset>
  RenewTime:       Thu, 09 May 2024 10:38:06 +0200
Conditions:
  Type                          Status  LastHeartbeatTime                 LastTransitionTime                Reason                          Message
  ----                          ------  -----------------                 ------------------                ------                          -------
  FrequentUnregisterNetDevice   False   Thu, 09 May 2024 10:35:17 +0200   Thu, 09 May 2024 10:00:13 +0200   NoFrequentUnregisterNetDevice   node is functioning properly
  SysctlChanged                 False   Thu, 09 May 2024 10:35:17 +0200   Thu, 09 May 2024 10:00:13 +0200   SysctlNotChanged                Default sysctls are in effect, no unexpected sysctl changes
  FrequentKubeletRestart        False   Thu, 09 May 2024 10:35:17 +0200   Thu, 09 May 2024 10:00:13 +0200   NoFrequentKubeletRestart        kubelet is functioning properly
  FrequentDockerRestart         False   Thu, 09 May 2024 10:35:17 +0200   Thu, 09 May 2024 10:00:13 +0200   NoFrequentDockerRestart         docker is functioning properly
  FrequentContainerdRestart     False   Thu, 09 May 2024 10:35:17 +0200   Thu, 09 May 2024 10:00:13 +0200   NoFrequentContainerdRestart     containerd is functioning properly
  KernelDeadlock                False   Thu, 09 May 2024 10:35:17 +0200   Thu, 09 May 2024 10:00:13 +0200   KernelHasNoDeadlock             kernel has no deadlock
  ReadonlyFilesystem            False   Thu, 09 May 2024 10:35:17 +0200   Thu, 09 May 2024 10:00:13 +0200   FilesystemIsNotReadOnly         Filesystem is not read-only
  CorruptDockerOverlay2         False   Thu, 09 May 2024 10:35:17 +0200   Thu, 09 May 2024 10:00:13 +0200   NoCorruptDockerOverlay2         docker overlay2 is functioning properly
  NetworkUnavailable            False   Thu, 09 May 2024 10:00:31 +0200   Thu, 09 May 2024 10:00:31 +0200   RouteCreated                    RouteController created a route
  MemoryPressure                False   Thu, 09 May 2024 10:33:50 +0200   Thu, 09 May 2024 10:00:07 +0200   KubeletHasSufficientMemory      kubelet has sufficient memory available
  DiskPressure                  False   Thu, 09 May 2024 10:33:50 +0200   Thu, 09 May 2024 10:00:07 +0200   KubeletHasNoDiskPressure        kubelet has no disk pressure
  PIDPressure                   False   Thu, 09 May 2024 10:33:50 +0200   Thu, 09 May 2024 10:00:07 +0200   KubeletHasSufficientPID         kubelet has sufficient PID available
  Ready                         True    Thu, 09 May 2024 10:33:50 +0200   Thu, 09 May 2024 10:00:21 +0200   KubeletReady                    kubelet is posting ready status. AppArmor enabled
Addresses:
  InternalIP:  10.132.0.63
  ExternalIP:  34.38.199.53
Capacity:
  cpu:                2
  ephemeral-storage:  26588420Ki
  hugepages-1Gi:      0
  hugepages-2Mi:      0
  memory:             16387048Ki
  pods:               110
Allocatable:
  cpu:                1930m
  ephemeral-storage:  8397760472
  hugepages-1Gi:      0
  hugepages-2Mi:      0
  memory:             13598696Ki
  pods:               110
System Info:
  Machine ID:                 021bb2b073d2e6a1a6be85e8a0814853
  System UUID:                021bb2b0-73d2-e6a1-a6be-85e8a0814853
  Boot ID:                    ad8b1c9a-8d9b-49fb-b334-b9d2c0e204d5
  Kernel Version:             6.1.58+
  OS Image:                   Container-Optimized OS from Google
  Operating System:           linux
  Architecture:               amd64
  Container Runtime Version:  containerd://1.7.10
  Kubelet Version:            v1.29.1-gke.1589018
  Kube-Proxy Version:         v1.29.1-gke.1589018
PodCIDR:                      10.24.1.0/24
PodCIDRs:                     10.24.1.0/24
ProviderID:                   gce://qcl-hub/europe-west1-d/gke-qcl-cluster-core-pool-f96ea250-oiq6
Non-terminated Pods:          (36 in total)
  Namespace                   Name                                                   CPU Requests  CPU Limits  Memory Requests  Memory Limits  Age
  ---------                   ----                                                   ------------  ----------  ---------------  -------------  ---
  cert-manager                cert-manager-58b759d979-7mkjg                          0 (0%)        0 (0%)      0 (0%)           0 (0%)         37m
  cert-manager                cert-manager-cainjector-86dcff788d-n9nws               0 (0%)        0 (0%)      0 (0%)           0 (0%)         37m
  cert-manager                cert-manager-webhook-8cf9f8df6-97zv8                   0 (0%)        0 (0%)      0 (0%)           0 (0%)         37m
  kube-system                 calico-node-68d8s                                      100m (5%)     0 (0%)      0 (0%)           0 (0%)         38m
  kube-system                 calico-node-vertical-autoscaler-67b556c745-95ql5       0 (0%)        0 (0%)      0 (0%)           0 (0%)         37m
  kube-system                 calico-typha-599f75c799-262hw                          200m (10%)    0 (0%)      0 (0%)           0 (0%)         37m
  kube-system                 calico-typha-horizontal-autoscaler-6fcbc59848-dj6f8    10m (0%)      0 (0%)      0 (0%)           0 (0%)         37m
  kube-system                 calico-typha-vertical-autoscaler-6979784b8b-2s8z6      0 (0%)        0 (0%)      0 (0%)           0 (0%)         37m
  kube-system                 event-exporter-gke-9d5f84494-z2hjd                     3m (0%)       0 (0%)      100Mi (0%)       0 (0%)         37m
  kube-system                 fluentbit-gke-sbg5l                                    105m (5%)     0 (0%)      230Mi (1%)       530Mi (3%)     38m
  kube-system                 gke-metadata-server-wwwmp                              100m (5%)     0 (0%)      100Mi (0%)       100Mi (0%)     38m
  kube-system                 ip-masq-agent-vvrt7                                    10m (0%)      0 (0%)      16Mi (0%)        0 (0%)         38m
  kube-system                 konnectivity-agent-688b985888-82lpb                    15m (0%)      0 (0%)      60Mi (0%)        155Mi (1%)     37m
  kube-system                 konnectivity-agent-688b985888-brrxh                    15m (0%)      0 (0%)      60Mi (0%)        155Mi (1%)     37m
  kube-system                 konnectivity-agent-autoscaler-77f84b86b7-lc4gx         10m (0%)      0 (0%)      10M (0%)         0 (0%)         37m
  kube-system                 kube-dns-6cfb78fbf6-bsw84                              260m (13%)    0 (0%)      110Mi (0%)       210Mi (1%)     37m
  kube-system                 kube-dns-6cfb78fbf6-dwc99                              260m (13%)    0 (0%)      110Mi (0%)       210Mi (1%)     37m
  kube-system                 kube-dns-autoscaler-79b96f5cb-gfnqq                    20m (1%)      0 (0%)      10Mi (0%)        0 (0%)         37m
  kube-system                 kube-proxy-gke-qcl-cluster-core-pool-f96ea250-oiq6     100m (5%)     0 (0%)      0 (0%)           0 (0%)         38m
  kube-system                 metrics-server-v0.6.3-586c77f697-jdzk2                 49m (2%)      44m (2%)    113Mi (0%)       363Mi (2%)     37m
  kube-system                 netd-rsbt2                                             7m (0%)       0 (0%)      60Mi (0%)        30Mi (0%)      38m
  kube-system                 pdcsi-node-9bcbg                                       10m (0%)      0 (0%)      20Mi (0%)        100Mi (0%)     38m
  prod                        hub-66f7c686c-96568                                    10m (0%)      0 (0%)      128Mi (0%)       2Gi (15%)      37m
  prod                        proxy-646dd5ff58-94lpt                                 10m (0%)      0 (0%)      64Mi (0%)        1Gi (7%)       37m
  prod                        shared-dirsize-metrics-5c4cccfb48-kh26p                10m (0%)      50m (2%)    128Mi (0%)       256Mi (1%)     37m
  prod                        shared-volume-metrics-6d54b49fb-54bln                  0 (0%)        0 (0%)      0 (0%)           0 (0%)         37m
  staging                     hub-74d96fc58d-p45xn                                   10m (0%)      0 (0%)      128Mi (0%)       2Gi (15%)      37m
  staging                     proxy-d79d6b-2gbbn                                     10m (0%)      0 (0%)      64Mi (0%)        1Gi (7%)       37m
  staging                     shared-dirsize-metrics-5c4cccfb48-fkzbj                10m (0%)      50m (2%)    128Mi (0%)       256Mi (1%)     37m
  staging                     shared-volume-metrics-6d54b49fb-6vdjt                  0 (0%)        0 (0%)      0 (0%)           0 (0%)         37m
  support                     support-cryptnono-g7xcd                                5m (0%)       800m (41%)  100Mi (0%)       2Gi (15%)      38m
  support                     support-grafana-67df6d7dd5-gbm54                       10m (0%)      100m (5%)   200Mi (1%)       400Mi (3%)     37m
  support                     support-ingress-nginx-controller-59d4ddbf4d-5vwg2      100m (5%)     0 (0%)      250Mi (1%)       0 (0%)         37m
  support                     support-kube-state-metrics-7b5d88d67f-kpw46            0 (0%)        0 (0%)      0 (0%)           0 (0%)         37m
  support                     support-prometheus-node-exporter-8q7pd                 10m (0%)      10m (0%)    30Mi (0%)        30Mi (0%)      38m
  support                     support-prometheus-server-5787dd845b-7n4rw             100m (5%)     4 (207%)    4Gi (30%)        4Gi (30%)      37m
Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
  Resource           Requests          Limits
  --------           --------          ------
  cpu                1559m (80%)       5054m (261%)
  memory             6621271680 (47%)  15083Mi (113%)
  ephemeral-storage  0 (0%)            0 (0%)
  hugepages-1Gi      0 (0%)            0 (0%)
  hugepages-2Mi      0 (0%)            0 (0%)
Events:
  Type     Reason                                  Age                From                               Message
  ----     ------                                  ----               ----                               -------
  Normal   Starting                                38m                kube-proxy                         
  Normal   Starting                                38m                kubelet                            Starting kubelet.
  Warning  InvalidDiskCapacity                     38m                kubelet                            invalid capacity 0 on image filesystem
  Normal   NodeHasSufficientMemory                 38m (x4 over 38m)  kubelet                            Node gke-qcl-cluster-core-pool-f96ea250-oiq6 status is now: NodeHasSufficientMemory
  Normal   NodeHasNoDiskPressure                   38m (x4 over 38m)  kubelet                            Node gke-qcl-cluster-core-pool-f96ea250-oiq6 status is now: NodeHasNoDiskPressure
  Normal   NodeHasSufficientPID                    38m (x4 over 38m)  kubelet                            Node gke-qcl-cluster-core-pool-f96ea250-oiq6 status is now: NodeHasSufficientPID
  Normal   NodeAllocatableEnforced                 38m                kubelet                            Updated Node Allocatable limit across pods
  Normal   Synced                                  38m                cloud-node-controller              Node synced successfully
  Normal   RegisteredNode                          38m                node-controller                    Node gke-qcl-cluster-core-pool-f96ea250-oiq6 event: Registered Node gke-qcl-cluster-core-pool-f96ea250-oiq6 in Controller
  Warning  NodeRegistrationCheckerStart            38m                node-registration-checker-monitor  Thu May  9 08:00:05 UTC 2024 - ** Starting Node Registration Checker **
  Normal   NodeReady                               37m                kubelet                            Node gke-qcl-cluster-core-pool-f96ea250-oiq6 status is now: NodeReady
  Warning  NodeRegistrationCheckerDidNotRunChecks  31m                node-registration-checker-monitor  Thu May  9 08:07:06 UTC 2024 - **     Node ready and registered. **
```

</details>

--- a/scripts/monero.bt
+++ b/scripts/monero.bt
@@ -19,12 +19,11 @@
 
 // See https://blog.px.dev/detect-monero-miners/ for a description of what
 // this program actually does.
-#include <asm/fpu/internal.h>
-#include <asm/fpu/types.h>
+#include <asm/fpu/api.h>
 tracepoint:x86_fpu:x86_fpu_regs_deactivated
 {
     $f = (struct fpu *)args->fpu;
-    $mxcsr = $f->state.xsave.i387.mxcsr;
+    $mxcsr = $f->__fpstate.regs.xsave.i387.mxcsr;
     $fpcr = ($mxcsr & 0x6000) >> 13;
     if ($fpcr != 0) {
 

